app/domain/authentication/validate_status.rb

```diff
@@ -0,0 +1,124 @@
+# frozen_string_literal: true
+
+require 'command_class'
+require 'authentication/webservices'
+
+module Authentication
+
+  Err = Errors::Authentication
+
+  # Possible Errors Raised:
+  # AuthenticatorNotFound, StatusNotImplemented, AccountNotDefined
+  # ServiceNotDefined, UserNotDefinedInConjur, UserNotAuthorizedInConjur,
+  # NotWhitelisted
+
+  ValidateStatus = CommandClass.new(
+    dependencies: {
+      validate_whitelisted_webservice: ::Authentication::Security::ValidateWhitelistedWebservice.new,
+      validate_webservice_access: ::Authentication::Security::ValidateWebserviceAccess.new,
+      validate_webservice_exists: ::Authentication::Security::ValidateWebserviceExists.new,
+      role_class: ::Role,
+      implemented_authenticators: Authentication::InstalledAuthenticators.authenticators(ENV),
+      enabled_authenticators: ENV['CONJUR_AUTHENTICATORS'],
+      audit_event: AuditEvent.new
+    },
+    inputs: %i(authenticator_name account authenticator_webservice user)
+  ) do
+
+    def call
+      validate_authenticator_exists
+      validate_authenticator_implements_status_check
+
+      validate_user_has_access_to_status_webservice
+
+      validate_authenticator_webservice_exists
+      validate_webservice_is_whitelisted
+
+      validate_authenticator_requirements
+
+      audit_success
+    rescue => e
+      audit_failure(e)
+      raise e
+    end
+
+    private
+
+    def validate_authenticator_exists
+      raise Err::AuthenticatorNotFound, @authenticator_name unless authenticator
+    end
+
+    def validate_authenticator_implements_status_check
+      raise Err::StatusNotImplemented, @authenticator_name unless authenticator.class.method_defined?(:status)
+    end
```

Comment:
Yeah, I kind of took advantage of the current design and did something similar to what we do with the `valid?` method. It is quite neat IMO actually. why do you think checking that a file exists under a folder is better than checking if a method exists under an authenticator class?

Reply:
I think it's ok to guide future devs that when they add a new authentication method they need to add a class with these requirements:

1. Class named `Authenticator`
1. Parent class named `AuthnSomething`
1. Grandparent class named `Authentication`
1. A method named `valid?`
1. Optional - If you implement a status check add a method named `status`

Reply:
> why do you think checking that a file exists under a folder is better than checking if a method exists under an authenticator class?

Good question.  I think mainly so that an authenticator can be nothing but a function, ie, something with a `call` method.  Which is both simpler and means that you can use a `CommandClass` directly, rather than having a special shell class that exists only to satisfy the required methods.  I created the whole `valid` thing before I started using Command classes.

Reply:
this looks like something that needs extra design. i'm afraid it will be delayed to sometime in the future.

Reply:
That's ok.  And I agree this needs more thought and design and proper design cycle.

I do think it's important, though, and deserves an issue.

Reply:
it kind of has already here: https://github.com/cyberark/conjur/issues/1053

I added a comment there for clarity.